indico/modules/events/registration/templates/management/regform_edit.html

```diff
@@ -29,4 +29,43 @@
             {% trans %}Back{% endtrans %}
         </a>
     {% endcall %}
+
+    {% if regform %}
+        <script>
+            (function () {
+                'use strict';
+
+                $('#message-pending-preview').attr('data-href', '{{ url_for('.notification_preview', regform) }}').ajaxDialog({
+                    title: $T('Pending Registration Preview'),
+                    method: 'POST',
+                    data: function () {
+                        return {
+                            message: $('#message_pending').val(),
+                            state: 'pending',
+                        };
+                    }
+                });
+                $('#message-unpaid-preview').attr('data-href', '{{ url_for('.notification_preview', regform) }}').ajaxDialog({
+                    title: $T('Unpaid Registration Preview'),
```

Comment:
Calling `$T()` is very legacy, Use `$T.gettext(...)` instead.

Reply:
IIRC you can also use data-title and data-method on the trigger element instead of specifying this here --> combined with my other suggestion this could be in the Python code, and make the JS here much much smaller

Reply:
`data-title` works, but when setting `data-method` I get a 422. I did the changes with everything but  that.

Reply:
I added another commit that goes full-declarative. Besides one small missing piece, everything was already supported anyway :)